android/app/src/main/java/org/openbot/robot/CameraActivity.java

```diff
@@ -682,6 +683,20 @@ public void onRequestPermissionsResult(
         }
         break;
 
+      case REQUEST_RECORD_PERMISSION_CONTROLLER:
+        if (grantResults.length > 0 && grantResults[0] == PackageManager.PERMISSION_GRANTED) {
+          // ask for recording permissions here
+
+          ActivityCompat.requestPermissions(
+              this, new String[] {RECORD_AUDIO}, REQUEST_RECORD_PERMISSION_CONTROLLER);
+
+        } else {
+          if (ActivityCompat.shouldShowRequestPermissionRationale(this, PERMISSION_LOCATION)) {
+            Toast.makeText(this, R.string.location_permission_denied_controller, Toast.LENGTH_LONG)
```

Comment:
There should be a new string defined for the audio permission, not display the location permission text.